app/javascript/chat/configMenu.jsx

```diff
@@ -0,0 +1,52 @@
+import { h, Component, createRef } from 'preact';
+// eslint-disable-next-line import/no-unresolved
+import ConfigImage from 'images/overflow-horizontal.svg';
+
+export class ConfigMenu extends Component {
+  constructor() {
+    super();
+    this.state = { visible: false };
+    this.firstNavLink = createRef();
+    this.configMenuButton = createRef();
+  }
+
+  handleClick = () => {
+    this.setState(
+      (prevState) => ({ visible: !prevState.visible }),
+      () => {
+        this.state.visible
+          ? this.firstNavLink.current.focus()
+          : this.configMenuButton.current.focus();
+      },
+    );
+  };
+
+  render() {
+    const { visible } = this.state;
+
+    return (
+      <div className="chatchannels__config">
+        <button
+          onClick={this.handleClick}
+          aria-label="toggle configuration menu"
+          style={{ backgroundImage: `url(${ConfigImage})` }}
+          ref={this.configMenuButton}
```

Comment:
(refactor/blocking) - This is working grand (and such a big improvement for accessibility!), but it'd be even better if we add `aria-expanded` so screen reader users know whether the menu is currently open/closed (and then we can remove the 'toggle' from the aria-label):

```suggestion
        <button
          onClick={this.handleClick}
          aria-expanded={visible}
          aria-label="configuration navigation menu"
          style={{ backgroundImage: `url(${ConfigImage})` }}
          ref={this.configMenuButton}
```